plugins/lookup/pre_flight.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import absolute_import, division, print_function
+import json
+from distutils.version import LooseVersion
+from pkg_resources import parse_requirements
+from six.moves.urllib.request import urlopen
+
+from ansible import constants as C
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_text, to_native
+from ansible.plugins.lookup import LookupBase
+from ansible.utils.display import Display
+
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+lookup: pre_flight
+author:
+  - Pulp Team (@pulp) <pulp-dev@redhat.com>
+version_added: "2.9"
+requirements:
+  - disutils
+  - json
+  - pkg_resources
+  - six
+short_description: Check if all plugins are compatible with the current pulpcore version.
+description:
+  - Check if all plugins are compatible with the current pulpcore version.
+options:
+  terms:
+    description: A list of Github termsitories from which to retrieve versions.
+    required: True
+notes:
+  - The version tag is returned however it is defined by the Github termsitory.
+  - Most termsitories used the convention 'vX.X.X' for a tag, while some use 'X.X.X'.
+  - Some may use release tagging structures other than semver.
+  - This plugin does not perform opinionated formatting of the release tag structure.
+  - Users should format the value via filters after calling this plugin, if needed.
+seealso:
+  - name: Github Releases API
+    description: API documentation for retrieving the latest version of a release.
+    link: https://developer.github.com/v3/terms/releases/#get-the-latest-release
+"""
+
+EXAMPLES = r"""
+- name: Strip the 'v' out of the tag version, e.g. 'v1.0.0' -> '1.0.0'
+  set_fact:
+    ansible_version: "{{ lookup('pulp.pulp_installer.pre_flight', 'ansible/ansible')[1:] }}"
+- name: Operate on multiple termsitories
+  git:
+    repo: https://github.com/{{ item }}.git
+    version: "{{ lookup('pulp.pulp_installer.pre_flight', item) }}"
+    dest: "{{ lookup('env', 'HOME') }}/projects"
+  with_items:
+    - ansible/ansible
+    - ansible/molecule
+    - ansible/awx
+"""
+
+RETURN = r"""
+  _list:
+    description:
+      - List of latest Github termsitory version(s)
```

Comment:
Is that a *thing*? Or did your `s/repos/terms/` go mad?

Reply:
I just copied https://github.com/artis3n/ansible-collection-github and used as placeholder, I modified the code, but I still have to work on the documentation part